bodhi/server/consumers/greenwave.py

```diff
@@ -0,0 +1,66 @@
+# Copyright © 2019 Red Hat, Inc.
+#
+# This file is part of Bodhi.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+The "greenwave handler".
+
+This module is responsible for listening for messages from greenwave.
+It then updates the policies of the build that greenwave checked.
+"""
+
+import logging
+
+import fedora_messaging
+
+from bodhi.server.models import Build
+from bodhi.server.util import transactional_session_maker
+from bodhi.server.exceptions import BodhiException
+
+log = logging.getLogger(__name__)
+
+
+class GreenwaveHandler:
+    """
+    The Bodhi Greenwave Handler.
+
+    A fedora-messaging listener waiting for messages from greenwave about enforced policies.
+    """
+
+    def __init__(self):
+        """Initialize the GreenwaveHandler."""
+        self.db_factory = transactional_session_maker()
+
+    def __call__(self, message: fedora_messaging.api.Message):
+        """Handle messages arriving with the configured topic."""
+        msg = message.body.get("msg")
+        if msg is None:
+            log.debug("Ignoring message without 'msg'.")
+            return
```

Comment:
With fedora-messaging instead of fedmsg, the interesting contents are directly in the body, see e.g. commit 884a0ccdc9f597e3eeb41442f9d5e71164ffb5e9. I.e., the above would simply be:

```
    msg = message.body
```

Ignoring messages without a `msg` block isn't necessary anymore.

Reply:
Although I do believe you, I'm going to wait until I capture an actual greenwave fedora-message before doing this change.  I thought I was working off a fedora-messager message, but I guess it was fedmsg.

Reply:
After todays meeting, I agree with you.  I will be changing all messaging functions to get things straight from body, and not body['msg']